app/assets/src/components/ui/controls/dropdowns/common/DropdownTrigger.jsx

```diff
@@ -1,9 +1,53 @@
 import React from "react";
 import cx from "classnames";
+import { get } from "lodash/fp";
+
 import PropTypes from "prop-types";
 import cs from "./dropdown_trigger.scss";
 
+// Set of component names that render labels that contain badge counts
+const DROPDOWN_LABEL_OWNERS_WITH_BADGE_COUNTS = new Set([
+  "ThresholdFilterDropdown",
+  "MultipleNestedDropdown",
+]);
+
 class DropdownTrigger extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.resizeObserver = null;
+    this.labelContainerRef = null;
+    this.labelRef = null;
+
+    this.state = {
+      hideDropdownLabel: false,
+    };
+  }
+
+  componentDidMount() {
+    if (this.labelContainerRef) {
+      this.resizeObserver = new ResizeObserver(this.handleFilterResize);
+      this.resizeObserver.observe(this.labelContainerRef.current);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.resizeObserver) {
+      this.resizeObserver.disconnect();
+    }
+  }
+
+  handleFilterResize = labelContainers => {
+    const labelContainerWidth = labelContainers[0].contentRect.width - 3.7; // -3.7 removes padding
```

Comment:
I would use `this.labelContainerRef.current.offsetWidth` here, just so that you're using the same method to fetch both widths. Generally better to use one approach instead of two if they're the same.

Also. it seems that the `3.7` and `1` values here are related to the padding. This seems to correspond to `margin-right: 5` for `.label` in the CSS. Maybe you can factor it out into a single variable with a comment (to make it a little cleaner):
```
// See dropdown_trigger.scss
const labelTextMarginRight = 5;
```